### PR TITLE
Daily report

### PR DESCRIPTION
**Pull request by DOM-C1**

### CODE REVIEW

File: README.md
Comment (DKWeeto) [inline README.md:1]: 💯

File: architecture-diagram.png
Comment (DKWeeto) [inline architecture-diagram.png:1]: we'll need to change the to the updated architecture diagram: 
arch_typo drawio

File: website/utils_db.py [hunk 1]
```diff
@@ -62,7 +58,7 @@ def setup_user_location(details, name, email, sub_newsletter, sub_alerts, conn)
     longitude, latitude = get_standard_long_lat(location_name)
     country_id = get_id('country', 'name', country, conn)
     if country_id == -1:
-        return render_template('cant_be_found_page.html')
+        return render_template('page_not_found.html')
 
     county_id = get_id('county', 'name', county, conn)
     if county_id == -1:
```
Comment (NMcKittie) [inline website/utils_db.py]: Hard coded value, maybe have the -1 as a named global variable if its a const or even just declare it within function

File: daily_report/report.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    async with aioboto3.Session().client('ses', aws_access_key_id=ENV['AWS_KEY'],
+                                         aws_secret_access_key=ENV['AWS_SKEY'],
+                                         region_name='eu-west-2') as ses:
+        tasks = [asyncio.create_task(send_email(ses, await format_forecast_report(df, email), email))
+                 for email in df['email'].unique()]
+        await asyncio.gather(*tasks)
+    conn.close()
+
+
+def get_db_connection(config: dict) -> connect:
```
Comment (DKWeeto) [inline daily_report/report.py:64]: Is this the right typehint? It returns a 'connection', not the 'connect' function itself. This comes from psycopg.extensions

File: daily_report/report.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        FROM forecast AS F
+        JOIN weather_code WC ON F.weather_code_id = WC.weather_code_id
+        JOIN weather_report WR ON F.weather_report_id = WR.weather_report_id
+    """
+    user_details = execute_query(conn, user_details_query)
+    forecast_details = execute_query(conn, forecast_query).drop_duplicates()
+    forecast_details = forecast_details.loc[:,
+                                            ~forecast_details.columns.duplicated()]
+
+    combined_df = pd.merge(user_details, forecast_details,
```
Comment (DKWeeto) [inline daily_report/report.py:105]: could this not be done in an sql query? It might be faster

**Review by DKWeeto — CHANGES_REQUESTED** [review]

File: architecture-diagram.png
Comment (DOM-C1) [reply architecture-diagram.png:1]: done

File: daily_report/report.py (hunk 2 above)
Comment (DOM-C1) [reply daily_report/report.py:64]: done

File: website/utils_db.py (hunk 1 above)
Comment (DOM-C1) [reply website/utils_db.py]: done

**Review by NMcKittie — APPROVED** [review]
Looks Good to me

**Review by NMcKittie — APPROVED** [review]

**Review by DKWeeto — APPROVED** [review]

**Pull request merged**